### PR TITLE
added markdown exercise

### PR DESCRIPTION
**Pull request by jsteinshouer**
Thought I would take a crack at adding the Markdown exercise. Hope this looks OK.

### CODE REVIEW

**Comment by bdw429s** [comment]
Hi @jsteinshouer , thanks for sending the first community pull! :tada:  

So, here's a few things:

- The `Markdown.cfc` file needs to be a "blank" CFC as this is the empty file/method that the students will be putting their code in.  The reference solution should be all within `Solution.cfc`.  I'm not quite sure what you've got going on here since both files have code, but it's different code.  See this example from another exercise: https://github.com/exercism/cfml/blob/master/exercises/word-count/WordCount.cfc
- The `SolutionTest.cfc` file should need to be changed from how it was originally generated by the task runner.  It inherits all the specs from the `MarkdownTest.cfc` that it extends.  The only line of code that needs to be in there is the one that overrides the creation of the `SUT` variable.  See this example from another exercise: https://github.com/exercism/cfml/blob/master/exercises/word-count/SolutionTest.cfc
- Can you explain some of the methods in your `Solution.cfc` file that don't seem to be in use at all such as `isBold()` and `isItalic()`?

**Comment by jsteinshouer** [comment]
Hi @bdw429s , thanks for the feedback.

 I understood from the [description](https://github.com/exercism/problem-specifications/blob/master/exercises/markdown/description.md) that this was a refactoring exercise where you start with code that works but is  hard to follow.

I tried to port the code from the [the php](https://github.com/exercism/php/blob/master/exercises/markdown/markdown.php) version to cfml. I also used some of the code from the [csharp](https://github.com/exercism/csharp/tree/master/exercises/markdown) version of the exercise as guide for the Solution. 

I removed the tests from the SolutionsTests.cfc file. I also removed the unused methods in Solution.cfc. I think I was using them at one point during my refactoring and forgot to take them out.

**Comment by bdw429s** [comment]
Ahh, sorry-- I didn't realize that exercise was a refactoring one.  In that case, perfect!  Thanks for fixing up the other items.
